### DISCUSSION TITLE
How to use Counter api to show the total count of meter in ui

### DISCUSSION THREAD

**Discussion by malaxiaolongxia** [apache/skywalking#7578] [Q&A] [answered]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
I want to record the count of users ,so I code like this 
```
import org.apache.skywalking.apm.toolkit.meter.Counter
Counter counter = MeterFactory.counter("total_users").build();
counter.increment(1);   //create user

// counter.increment(-1);     //delete user
```
then i want to config a chart in ui to show total users  but i can only  config the number of new users per minute 
```
{
            "width":3,
            "title":"Title",
            "height":350,
            "entityType":"ServiceInstance",
            "independentSelector":false,
            "metricType":"REGULAR_VALUE",
            "metricName":"meter_total_users",
            "queryMetricType":"readMetricsValues",
            "chartType":"ChartLine"
        }
```
so how to config or code ？

**Comment by wu-sheng** [comment, marked as answer]
You need to create meter script first, please follow this doc, https://skywalking.apache.org/docs/main/v8.7.0/en/setup/backend/backend-meter/#meters-configuration. And don't forget to learn MAL.

We have a lot of MAL examples already, you could learn from it.

**Reply by malaxiaolongxia** [reply]
ok thinks a lot，i solve this by create meter script like this
```
expSuffix: instance(['service'], ['instance'])
metricPrefix: meter
metricsRules:
  - name: total_users
    exp: total_users.sum(['service', 'instance'])
``` 
i have another question ....
a lot of data miss because i don't report meter when the skywalking didn't work correctly a long time ago , can i report the data  again

**Reply by wu-sheng** [reply]
That depends on what do you mean `report again`. Because you can't set the timestamp actually.

**Reply by malaxiaolongxia** [reply]
i mean that any api can help me to report meter that happened before?

**Reply by wu-sheng** [reply]
In that case, there isn't. In typical observability use case, if backend service is not available, then, the metrics should be ignored.
All backend streaming and even database are designed for latest and hot data only.

**Reply by malaxiaolongxia** [reply]
i think this is important in business dashboard，it is not like jvm,cpu,io..
eg: when i create a meter means 'order count',i will always make sure the count is correct
- when a new meter be added, historical data should be added to ensure business integrity
-  if backend service is not available, business data should be repaired

**Reply by wu-sheng** [reply]
No, it isn't. You should not take SkyWalking or any APM like a BI system.
The counter will be eventually correct when the backend back to work. But don't expect to keep historical data always correct. If you want, try to make the backend stable. 
That API wouldn't be provided, no matter from design and technology possibility perspectives.

**Reply by wu-sheng** [reply]
The APM expectation is that, the data would be x,0,0,0,y. The value Y is correct.

**Reply by malaxiaolongxia** [reply]
ok, thanks a lot.
